### ISSUE TITLE
oneOf/anyOf schema can't enter in data with omitExtraData and liveOmit

### ISSUE THREAD

**Issue by FunJim** [rjsf-team/react-jsonschema-form#1628]
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

When setting `omitExtraData` and `liveOmit` prop to true, form field can't be filled in with data.

### Steps to Reproduce

1. Go to [rjsf playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0Iiwib25lT2YiOlt7InByb3BlcnRpZXMiOnsibG9yZW0iOnsidHlwZSI6InN0cmluZyJ9fSwicmVxdWlyZWQiOlsibG9yZW0iXX0seyJwcm9wZXJ0aWVzIjp7Imlwc3VtIjp7InR5cGUiOiJzdHJpbmcifX0sInJlcXVpcmVkIjpbImlwc3VtIl19XX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjp0cnVlLCJkaXNhYmxlIjpmYWxzZSwib21pdEV4dHJhRGF0YSI6dHJ1ZSwibGl2ZU9taXQiOnRydWV9fQ==). The `oneOf` schema is provided by the playground
2. Notice that `Omit extra data` and `Live omit` are set to true.
3. Try to fill in the form field and nothing can be entered in.
4. Same thing if you choose `anyOf` schema.

#### Expected behavior

Data can be filled in when using oneOf/anyOf schema with omitExtraData and liveOmit.

### Version
Current rjsf playground
https://rjsf-team.github.io/react-jsonschema-form/

**Comment by AlimovSV** [comment]
Looks like the problem is in the `toPathSchema` method, where there is no logic for anyOf/oneOf cases. Try this patch - it works for me:
```
diff --git a/node_modules/@rjsf/core/dist/cjs/utils.js b/node_modules/@rjsf/core/dist/cjs/utils.js
index 138e5d8..196f7a6 100644
--- a/node_modules/@rjsf/core/dist/cjs/utils.js
+++ b/node_modules/@rjsf/core/dist/cjs/utils.js
@@ -1141,6 +1141,14 @@ function toPathSchema(schema) {
       // array item has just been added, but not populated with data yet
       (formData || {})[property]);
     }
+  } else if (Array.isArray(schema.anyOf)) {
+    schema.anyOf.forEach((_schema) => {
+      Object.assign(pathSchema, toPathSchema(_schema, name, rootSchema, formData));
+    });
+  } else if (Array.isArray(schema.oneOf)) {
+    schema.oneOf.forEach((_schema) => {
+      Object.assign(pathSchema, toPathSchema(_schema, name, rootSchema, formData));
+    });
   }
 
   return pathSchema;
```

**Comment by bysph** [comment]
@epicfaace  
Hi, is there a fix for this issue?

**Comment by ksbrar** [comment]
bump, any known workaround?

**Comment by benjdlambert** [comment]
@AlimovSV maybe you can raise this in a PR for approval and merge?

**Comment by AlimovSV** [comment]
An additional example of the wrong behavior liveOmit/omitExtraData. When array field identifier contains `.`:

Steps to Reproduce
---
- Go to [rjsf playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJhLmIiOlt7fV19LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsiYS5iIjp7InR5cGUiOiJhcnJheSIsIml0ZW1zIjp7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7Im5hbWUiOnsidHlwZSI6InN0cmluZyJ9fX0sIm1pbkl0ZW1zIjoxfX19LCJ1aVNjaGVtYSI6e30sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOnRydWUsImxpdmVPbWl0Ijp0cnVlfX0=). 
- Notice that Omit extra data and Live omit are set to true.
- Try to fill in the form field and nothing can be entered in.

**Comment by AlimovSV** [comment]
Looks like v5 has fixed the previous example but found another one: [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJhIjpmYWxzZX0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJhIjp7ImNvbnN0Ijp0cnVlLCJkZWZhdWx0IjpmYWxzZX19LCJpZiI6eyJwcm9wZXJ0aWVzIjp7ImEiOnsiY29uc3QiOnRydWV9fX0sInRoZW4iOnsicHJvcGVydGllcyI6eyJncm91cCI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJiIjp7InR5cGUiOiJib29sZWFuIiwiZGVmYXVsdCI6ZmFsc2UsImNvbnN0Ijp0cnVlfX0sImlmIjp7InByb3BlcnRpZXMiOnsiYiI6eyJjb25zdCI6dHJ1ZX19LCJyZXF1aXJlZCI6WyJiIl19LCJ0aGVuIjp7InByb3BlcnRpZXMiOnsiYyI6eyJ0eXBlIjoiYm9vbGVhbiIsImRlZmF1bHQiOmZhbHNlLCJjb25zdCI6dHJ1ZX19fX19fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InNob3dFcnJvckxpc3QiOiJ0b3AiLCJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOnRydWUsImxpdmVPbWl0Ijp0cnVlLCJmb2N1c09uRmlyc3RFcnJvciI6ZmFsc2UsImxpdmVWYWxpZGF0ZSI6dHJ1ZSwibm9WYWxpZGF0ZSI6dHJ1ZX19)

**Comment by heath-freenome** [comment]
> Looks like v5 has fixed the previous example but I found yet another one: [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJhIjpmYWxzZX0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJhIjp7ImNvbnN0Ijp0cnVlLCJkZWZhdWx0IjpmYWxzZX19LCJpZiI6eyJwcm9wZXJ0aWVzIjp7ImEiOnsiY29uc3QiOnRydWV9fX0sInRoZW4iOnsicHJvcGVydGllcyI6eyJncm91cCI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJiIjp7InR5cGUiOiJib29sZWFuIiwiZGVmYXVsdCI6ZmFsc2UsImNvbnN0Ijp0cnVlfX0sImlmIjp7InByb3BlcnRpZXMiOnsiYiI6eyJjb25zdCI6dHJ1ZX19LCJyZXF1aXJlZCI6WyJiIl19LCJ0aGVuIjp7InByb3BlcnRpZXMiOnsiYyI6eyJ0eXBlIjoiYm9vbGVhbiIsImRlZmF1bHQiOmZhbHNlLCJjb25zdCI6dHJ1ZX19fX19fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InNob3dFcnJvckxpc3QiOiJ0b3AiLCJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOnRydWUsImxpdmVPbWl0Ijp0cnVlLCJmb2N1c09uRmlyc3RFcnJvciI6ZmFsc2UsImxpdmVWYWxpZGF0ZSI6dHJ1ZSwibm9WYWxpZGF0ZSI6dHJ1ZX19). In this example I can't check the "c" checkbox until liveOmit is true
> 
> UPD: Found the [solution](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJhIjpmYWxzZX0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJhIjp7ImNvbnN0Ijp0cnVlLCJkZWZhdWx0IjpmYWxzZX19LCJpZiI6eyJwcm9wZXJ0aWVzIjp7ImEiOnsiY29uc3QiOnRydWV9fX0sInRoZW4iOnsicHJvcGVydGllcyI6eyJncm91cCI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJiIjp7InR5cGUiOiJib29sZWFuIiwiZGVmYXVsdCI6ZmFsc2UsImNvbnN0Ijp0cnVlfX0sImFsbE9mIjpbeyJpZiI6eyJwcm9wZXJ0aWVzIjp7ImIiOnsiY29uc3QiOnRydWV9fSwicmVxdWlyZWQiOlsiYiJdfSwidGhlbiI6eyJwcm9wZXJ0aWVzIjp7ImMiOnsidHlwZSI6ImJvb2xlYW4iLCJkZWZhdWx0IjpmYWxzZSwiY29uc3QiOnRydWV9fX19XX19fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InNob3dFcnJvckxpc3QiOiJ0b3AiLCJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOnRydWUsImxpdmVPbWl0Ijp0cnVlLCJmb2N1c09uRmlyc3RFcnJvciI6ZmFsc2UsImxpdmVWYWxpZGF0ZSI6dHJ1ZSwibm9WYWxpZGF0ZSI6dHJ1ZX19)

Can we close this then?

**Comment by AlimovSV** [comment]
The mentioned solution doesn't work properly too. If you check the schema you will see a default: false for the "c" prop. But when you check the "b" prop, the "c" prop stays undefined.

**Comment by heath-freenome** [comment]
> The mentioned solution doesn't work properly too. If you check the schema you will see a default: false for the "c" prop. But when you check the "b" prop, the "c" prop stays undefined.

Are you wanting the value to be entirely removed? Isn't undefined considered no value?

**Comment by AlimovSV** [comment]
| Schema | UI | Data |
|---| ---|---|
| <img width="202" src="https://user-images.githubusercontent.com/4921904/222795372-1b49212d-3892-4c89-a65d-fe590b42533d.png"> | <img width="112" src="https://user-images.githubusercontent.com/4921904/222795368-767518c7-048b-4124-9900-34ec4928f6b7.png"> | <img width="141" src="https://user-images.githubusercontent.com/4921904/222795363-7117b098-e4fa-4a75-acc3-55fb90a9d4a0.png"> |

As you can see, the schema has "c".default = false, UI renders the "c" checkbox, but data doesn't contain any value for "c"

**Comment by heath-freenome** [comment]
Oops, this flag shouldn't have been there

**Comment by heath-freenome** [comment]
Fixed by #3627
